### PR TITLE
Add user put and post endpoints for neo4j

### PR DESCRIPTION
**Pull request by alran**
### Summary of Changes

This PR adds PUT and POST endpoints for `/users`. It makes use of the common BaseAPI. I also added a `LocalTransaction` class, which can be used to DRY up existing linking and upsert code as follow up. 

I tested these endpoints out manually like:
```
curl -X POST http://0.0.0.0:5002/user -H "Content-Type: application/json" -d '[{"email": "y@gmail.com"}]'
curl -X PUT http://0.0.0.0:5002/user -H "Content-Type: application/json" -d '{"email": "xxx@gmail.com"}'
```

### Tests
I added tests for the put_user and post_users methods for the neo4j_proxy. I also added tests for the put and post on the UserDetailAPI

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [x] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [x] PR includes a summary of changes. 
- [x] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [x] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public functions and the classes in the PR contain docstrings that explain what it does
- [x] PR passes `make test`

### CODE REVIEW

File: tests/unit/api/test_user.py [hunk 1]
```diff
@@ -28,6 +28,22 @@ def test_gets(self) -> None:
         self.assertEqual(list(response)[1], HTTPStatus.OK)
         self.mock_client.get_users.assert_called_once()
 
+    @mock.patch('metadata_service.api.request')
+    def test_post(self, mock_request: MagicMock) -> None:
+        mock_request.get_json.return_value = [dict(email='username@gmail.com', is_active=True)]
```
Comment (friendtocephalopods) [inline tests/unit/api/test_user.py]: tiny nit: maybe example.com so it won't resolve to any real user?

Comment (alran) [reply tests/unit/api/test_user.py]: fair enough. I updated to use example.com email addresses

**Comment by dmateusp** [comment]
apart from the conflicting files, I think this is missing swagger docs https://github.com/lyft/amundsenmetadatalibrary/tree/master/metadata_service/api/swagger_doc

**Comment by alran** [comment]
let me make some updates to address the comments!

File: metadata_service/proxy/neo4j_proxy.py [hunk 2]
```diff
@@ -26,6 +28,210 @@
...
+    TWO_TO_ONE = 2
+
+
+class Query:
+    def __init__(self, *, statement: str, params: Optional[Dict[str, Any]] = None) -> None:
+        self.statement: str = statement
+        self.params: Dict[str, Any] = params or {}
+
+
+class LocalTransaction:
```
[the remark this answers is not in the document]
Comment (alran) [reply metadata_service/proxy/neo4j_proxy.py]: Sure, I went back and updated other put/post/add methods to use LocalTransaction

**Comment by alran** [comment]
I think I have responded to all outstanding comments. Please let me know if I've missed anything!

**Pull request closed without merging**